tests/end2end/utils.py

```diff
@@ -0,0 +1,221 @@
+    if pdb:
+        kwargs = {}
+        p_args.insert(1, '--pdb')
+    else:
+        kwargs = {
+            'stdout': subprocess.PIPE,
+            'stderr': subprocess.PIPE,
+        }
+
+    # pylint: disable=consider-using-with
+    proc = subprocess.Popen(p_args, env=env, **kwargs)
+
+    stdout, stderr = proc.communicate()
+    rc = proc.returncode
+    if six.PY3 and not pdb:
+        stdout = stdout.decode()
+        stderr = stderr.decode()
+
+    return rc, stdout, stderr
+
+
+def _res_name(item):
+    """Return the resource name, used by pick_test_resources()"""
+    if isinstance(item, (tuple, list)):
+        return item[0].name
+    return item.name
+
+
+def pick_test_resources(res_list):
+    """
+    Return the list of resources to be tested.
+
+    The env.var "TESTRESOURCES" controls which resources are picked for the
+    test, as follows:
+
+    * 'random': (default) one random choice from the input list of resources.
+    * 'all': the complete input list of resources.
+    * '<pattern>': The resources with names matching the regexp pattern.
+
+    Parameters:
+      res_list (list of zhmcclient.BaseResource or tuple thereof):
+        List of resources to pick from. Tuple items are a resource and its
+        parent resources.
+
+    Returns:
+      list of zhmcclient.BaseResource: Picked list of resources.
+    """
+
+    test_res = os.getenv('TESTRESOURCES', 'random')
+
+    if test_res == 'random':
+        return [random.choice(res_list)]
+
+    if test_res == 'all':
+        return sorted(res_list, key=_res_name)
+
+    # match the pattern
+    ret_list = []
+    for item in res_list:
+        if re.search(test_res, _res_name(item)):
+            ret_list.append(item)
+    return sorted(ret_list, key=_res_name)
+
+
+def skipif_no_storage_mgmt_feature(cpc):
+    """
+    Skip the test if the "DPM Storage Management" feature is not enabled for
+    the specified CPC, or if the CPC does not yet support it.
+    """
+    try:
+        smf = cpc.feature_enabled('dpm-storage-management')
+    except ValueError:
+        smf = False
+    if not smf:
+        skip_warn("DPM Storage Mgmt feature not enabled or not supported "
+                  "on CPC {c}".format(c=cpc.name))
+
+
+def skipif_storage_mgmt_feature(cpc):
+    """
+    Skip the test if the "DPM Storage Management" feature is enabled for
+    the specified CPC.
+    """
+    try:
+        smf = cpc.feature_enabled('dpm-storage-management')
+    except ValueError:
+        smf = False
+    if smf:
+        skip_warn("DPM Storage Mgmt feature enabled on CPC {c}".
+                  format(c=cpc.name))
+
+
+def standard_partition_props(cpc, part_name):
+    """
+    Return the input properties for a standard partition in the specified CPC.
+    """
+    part_input_props = {
+        'name': part_name,
+        'description': 'Test partition for zhmcclient end2end tests',
+        'initial-memory': 1024,
+        'maximum-memory': 2048,
```

Comment:
You might want to update the max to at least 4 GB, if you ever intend to actually start such a partition (for example with the sample image), then 2 GB won't do.

Reply:
This method was copied from zhmcclient and is not used so far. I have removed it.